### PR TITLE
[7.x] Only replace methodName prefix if would be empty

### PR DESCRIPTION
**Pull request by sethmlarson**

### CODE REVIEW

**Review by Mpdreamz — APPROVED** [review]
LGTM

File: internal/cmd/generate/commands/gentests/skips.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ var skipFiles = []string{
 	"ml/evaluate_data_frame.yml", // Floats as map keys
 
 	"watcher/stats/10_basic.yml", // Sets "emit_stacktraces" as string ("true"), not bool
+	
+	"search.highlight/20_fvh.yml", // bad backslash
```
Comment (Mpdreamz) [inline internal/cmd/generate/commands/gentests/skips.go:35]: This particular change might need backporting to other branches?

**Comment by sethmlarson** [comment]
So ML integration tests are failing in this PR for reasons I can't figure out without more knowledge on debugging Go CI, however most of the API integration tests (specifically the ones we're adding here) are running and passing so I'm going to merge this now to get a valid `7.11` branch going.

File: internal/cmd/generate/commands/genstruct/command.go [hunk 2]
```diff
@@ -239,6 +239,11 @@ type API struct {
 			name := strings.ReplaceAll(e.Name(), "Request", "")
 			if strings.HasPrefix(strings.ToLower(name), strings.ToLower(n)) {
 				methodName := strings.ReplaceAll(name, n, "")
+				// Some methods are equal to the namespace (like 'rollup.rollup')
```
Comment (Mpdreamz) [inline internal/cmd/generate/commands/genstruct/command.go:242]: These changes would need a backport to other active integration branches too I assume?

File: internal/cmd/generate/commands/gentests/skips.go (hunk 1 above)
Comment (sethmlarson) [reply internal/cmd/generate/commands/gentests/skips.go:35]: Yeah, I'll get it backported to all the relevant branches.

**Pull request merged**